### PR TITLE
Feature/teleop

### PR DESCRIPTION
**Pull request by bgrayd**
I have started the teleop node, but it still needs the service calls to services that haven't been written yet.  It also needs to do the arm control, but I am not sure how defined that is yet.  This should also be able to be used in gazebo.

### CODE REVIEW

File: cr17/nodes/cr17/joystick_controller.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        # Load topic names
+        self.joystick_topic       = rospy.get_param("topics/joystick", "joy")
+        drive_topic               = rospy.get_param("topics/drive_cmds", "cmd_vel")
+        # Setup publishers
+        self.drive_pub = rospy.Publisher(drive_topic, Twist, queue_size = 10)
+        # Setup subscribers
+        rospy.Subscriber(self.joystick_topic, Joy, self.joy_callback)
+
+    def startTeleop(self):
+        #TODO: call the serives from navigator, scoop_controller, and the HLSC
```
Comment (JacobAMason) [inline cr17/nodes/cr17/joystick_controller.py]: "serives" ?

File: cr17/nodes/cr17/joystick_controller.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        self.teleopEnabled = True
+
+    def endTeleop(self):
+        #TODO: call the serives from navigator, scoop_controller, and the HLSC
```
Comment (JacobAMason) [inline cr17/nodes/cr17/joystick_controller.py]: "serives"

File: cr17/nodes/cr17/joystick_controller.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+            twister.linear.x = lin_hist_avg     #reduce sudden changes
+            twister.angular.z = vel_hist_avg    #reduce sudden changes
+            if self.teleopEnabled:
+                self.drive_pub.publish(twister)
+
+            self.joy_received = False
+            rate.sleep()
+
+    def _exit_handler(self):
```
Comment (JacobAMason) [inline cr17/nodes/cr17/joystick_controller.py]: Is this method ever used?

**Review by JacobAMason — CHANGES_REQUESTED** [review]
startTeleop and endTeleop are camel case functions and all your other functions are underscored.

**Comment by bgrayd** [comment]
I had not seen the comment about the function names, that has been fixed.  I also added in the service calls.

**Review by JacobAMason — APPROVED** [review]

**Pull request merged**